pkgs/by-name/pi/pipet/package.nix

```diff
@@ -0,0 +1,39 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+}:
+
+let
+  version = "0.2.1";
+in
+buildGoModule rec {
+  pname = "pipet";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "bjesus";
+    repo = "pipet";
+    rev = version;
+    hash = "sha256-PqOx/aFI5gHt78th1nkSKlTGw/r1eU7Ggz5kvtjMCmI=";
+  };
+
+  vendorHash = "sha256-jNIjF5jxcpNLAjuWo7OG/Ac4l6NpQNCKzYUgdAoL+C4=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.currentSha=${src.rev}"
+  ];
+
+  doCheck = false;
+
```

Comment:
Looks like it can generate completion? https://github.com/bjesus/pipet/blob/1f1469953dd426fc9e33f33e1472618390e155bd/cmd/pipet/main.go#L30

Try something like:

```
postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
  installShellCompletion --cmd pipet \
      --bash <($out/bin/pipet completion bash)
'';
```

This needs `installShellFiles` in `nativeBuildInputs`.

Reply:
Yes, please check out https://github.com/urfave/cli/blob/main/docs/v2/examples/bash-completions.md#distribution-and-persistent-autocompletion